### PR TITLE
DEVX-2221: ccloud_library cleans up ccloud usage for ksqldb metadata

### PR DESCRIPTION
**Pull request by rspurgeon**
### Description 

Uses JSON processing for ksqlDB metadata in ccloud library
https://confluentinc.atlassian.net/browse/DEVX-2221

### Author Validation
- [x] cp-quickstart (cloud)

### Reviewer Tasks

- [ ] cloud-etl (if setup to validate)

### CODE REVIEW

File: utils/ccloud_library.sh [hunk 1]
```diff
@@ -522,30 +522,37 @@ function ccloud::validate_ccloud_config() {
...
+	}
+
+	[ $# -gt 1 ] && echo "WARN: ccloud::validate_ksqldb_up function expects one parameter"
+
+  local ksqldb_endpoint=$1
 
   ccloud::validate_logged_in_ccloud_cli || exit 1
 
   if [[ "$ksqldb_endpoint" == "" ]]; then
     echo "ERROR: Provision a ksqlDB cluster via the Confluent Cloud UI and add the configuration parameter ksql.endpoint and ksql.basic.auth.user.info into your Confluent Cloud configuration file at $ccloud_config_file and try again."
```
Comment (ybyzek) [inline utils/ccloud_library.sh]: `ccloud_config_file` was removed as an argument, so is the value known in this function to be able to print out the error message?

**Comment by ybyzek** [comment]
Error on run:

```
[9:48:53] ~/git/examples/ccloud/ccloud-stack(DEVX-2221): ./ccloud_stack_create.sh 
This example uses real Confluent Cloud resources.
To avoid unexpected charges, carefully evaluate the cost of resources before launching the script and ensure all resources are destroyed after you are done running it.
Do you still want to run this script? [y/n] y
Do you also want to create a Confluent Cloud ksqlDB app (hourly charges may apply)? [y/n] y

Creating Confluent Cloud stack for service account demo-app-31277, ID: 132551.
Error: accepts 1 arg(s), received 29
Usage:
  ccloud kafka cluster describe <id> [flags]

Flags:
  -o, --output string   Specify the output format as "human", "json", or "yaml". (default "human")

Global Flags:
  -h, --help            Show help for this command.
  -v, --verbose count   Increase verbosity (-v for warn, -vv for info, -vvv for debug, -vvvv for trace).

Error: resource "Error" not found

Suggestions:
    Check that the resource "Error" exists.
    To list Kafka clusters, use `ccloud kafka cluster list`.
    To check schema-registry cluster info, use `ccloud schema-registry cluster describe`.
    To list KSQL clusters, use `ccloud ksql app list`.

Waiting up to 720 seconds for Confluent Cloud cluster to be ready and for credentials to propagate
.^C
```

File: utils/ccloud_library.sh (hunk 1 above)
Comment (rspurgeon) [reply utils/ccloud_library.sh]: Sorry, can you clarify the question?  My idea with the warning is just to provide an indiciation of improper invocation of the function.

**Comment by rspurgeon** [comment]
> Error on run:
> 
@ybyzek can you indicate your `jq` version?

File: utils/ccloud_library.sh (hunk 1 above)
Comment (rspurgeon) [reply utils/ccloud_library.sh]: I understand now, sorry the GitHub UI confused me with which line you were indicating.  Thanks for catching this, will address

Comment (rspurgeon) [reply utils/ccloud_library.sh]: removed in latest

**Review by ybyzek — CHANGES_REQUESTED** [review]
IIUC this PR makes `validate_credentials_ksqldb` no longer part of the validation workflow.  This function was useful because some people try to use Kafka credentials instead of creating new ksqlDB credentials.  So it is recommended to be part of validation somehow; it can be separate from checking if the endpoint is up, but it should be part of ksqlDB validation overall

**Comment by rspurgeon** [comment]
> IIUC this PR makes `validate_credentials_ksqldb` no longer part of the validation workflow. This function was useful because some people try to use Kafka credentials instead of creating new ksqlDB credentials. So it is recommended to be part of validation somehow; it can be separate from checking if the endpoint is up, but it should be part of ksqlDB validation overall

Thanks.  In an attempt to separate concerns and make the function more prescriptive to it's name, I did break this out.  Where the ksqlDB up check was called in external scripts, I added a additional call to the creds validation.

I'm not totally clear on the use cases you're describing that i've left out.  I've added this commit: https://github.com/confluentinc/examples/pull/855/commits/110da8d0555d65829c5ad01793ff2c8e891380ef#diff-5ebe849b3ac2695b290a288871dc904e950b16c2b0051e9d66555c4345143896R823 which adds calling the validate creds function to `function ccloud::validate_ccloud_stack_up()`, does this address your concern?  If not, would you mind adding some details on the use cases you're describing so I can add coverage for them?

**Comment by ybyzek** [comment]
@rspurgeon since a few `ccloud-stack` commits have been merged since this PR was opened, can you please rebase this to latest `6.0.0-post` and we'll give it a final run-thru before merging?

**Comment by rspurgeon** [comment]
@ybyzek Rebased and verified with ccloud 1.20.1

**Comment by ybyzek** [comment]
@rspurgeon FYI the current PR shows 31 modified files, including those not intended to be modified.  Maybe need rebase again or close this PR and open a new one?

**Comment by rspurgeon** [comment]
> @rspurgeon FYI the current PR shows 31 modified files, including those not intended to be modified. Maybe need rebase again or close this PR and open a new one?

Yikes, how did I do this now... thank you for catching.  I'll figure out what I did and repost

**Comment by rspurgeon** [comment]
closed in favor of #870

**Pull request closed without merging**